Plugins/NativeCamera/Source/Apple/NativeCameraImpl.mm

```diff
@@ -58,69 +61,147 @@ - (id)init:(std::shared_ptr<Babylon::Plugins::Camera::Impl::ImplData>)implData;
     {
     }
 
-    arcana::task<void, std::exception_ptr> Camera::Impl::Open(uint32_t /*width*/, uint32_t /*height*/, bool frontCamera)
+    arcana::task<void, std::exception_ptr> Camera::Impl::Open(uint32_t maxWidth, uint32_t maxHeight, bool frontCamera)
     {
+        if (maxWidth == 0 || maxWidth > std::numeric_limits<int32_t>::max()) {
+            maxWidth = std::numeric_limits<int32_t>::max();
```

Comment:
Why int32_t instead of uint32_t?

Reply:
It's an unreasonable resolution either way.  I'm giving overhead to prevent overflow because we might add both maxWidth and maxHeight on line 135 below:
                    uint32_t dimDiff{(maxWidth - dimensions.width) + (maxHeight - dimensions.height)};

If Apple ever releases a camera with 4294967295 x 4294967295 resolution we might be in trouble!